File: 55.跳跃游戏.go
/*
 * @lc app=leetcode.cn id=55 lang=golang
 *
 * [55] 跳跃游戏
 */

// @lc code=start
func canJump(nums []int) bool {
	maxIndex := 0 // 能到达的最远位置
	for idx, val := range nums {
		// 当前 idx 如果能够到达，同时从这个位置跳能超过 maxIndex 重置最远索引
		if maxIndex >= idx && idx+val > maxIndex {
			maxIndex = idx + val
		}
	}
	// 是否能够到达最后
	return maxIndex >= len(nums)-1
}

// @lc code=end

File: 55.跳跃游戏.py
#
# @lc app=leetcode.cn id=55 lang=python3
#
# [55] 跳跃游戏
#

# @lc code=start
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_index = 0       # 初始化当前能到达最远的位置
        for i, jump in enumerate(nums):   # i 为当前位置，jump 是当前位置的跳数
            if max_index >= i and i + jump > max_index:  # 如果当前位置能到达，并且当前位置 + 跳数 > 最远位置
                max_index = i + jump  # 更新最远能到达位置
        return max_index >= len(nums) - 1
# @lc code=end
